google/cloud/spanner/bytes.h

```diff
@@ -0,0 +1,157 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_BYTES_H_
+#define GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_BYTES_H_
+
+#include "google/cloud/spanner/version.h"
+#include "google/cloud/status_or.h"
+#include <array>
+#include <cstddef>
+#include <cstring>
+#include <iterator>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+
+/**
+ * A representation of the Spanner BYTES type ... variable-length binary data.
+ *
+ * A `Bytes` value can be constructed from, and converted to (1) a sequence of
+ * octets, or (2) a base64-encoded US-ASCII `std::string`. The base64 alphabet
```

Comment:
I don't think we want to expose anything about base64 in this class's interface. I think the base64 encoding should be completely an implementation detail hidden within this class.

If we need the ability to inject/extract the raw base64 string, we can expose internal friend functions that we can use, but users cannot.

Thoughts?

Reply:
The idea is that the user, as well as having their binary data in a variety of containers, might also have/want it as a base64-encoded string.  So, rather converting back-and-forth again, we can supply it directly.

Reply:
PTAL

Reply:
Now that the base64 functions are gone from the user-facing API, we should probably remove all reference to base64 in the documentation.